common/txmgr/types/tx.go

```diff
@@ -338,6 +338,15 @@ func (e *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetChecker() (Transm
 	return t, nil
 }
 
+func (e *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) IsPurged() bool {
```

Comment:
nit: I feel `IsPurged` is a bit misleading as it implies the tx has already been purged - confirmed on-chain with an empty attempt.